itemlist.lua

```diff
@@ -0,0 +1,223 @@
+	end
+	if minetest.is_player(player) then
+		return player:get_player_name()
+	end
+	minetest.log('warning',	'['..modname..'] get_player_name could not identify player.')
+end
+
+minetest.register_on_leaveplayer(function(player)
+	local playername = get_player_name(player)
+	if playername then
+		contexts[playername] = nil
+	end
+end)
+
+local function get_context(player)
+	local playername = get_player_name(player)
+	if playername then
+		if not contexts[playername] then
+			contexts[playername] = { playername = playername }
+		end
+		return contexts[playername]
+	end
+end
+
+-- Show node formspec functions
+local function show_node_formspec(playername, pos)
+	local meta = minetest.get_meta(pos)
+
+	-- Decontextualize formspec
+	local fs = meta:get_string('formspec')
+
+	-- Change context and currrent_name references to nodemeta references
+	local nodemeta = string.format("nodemeta:%i,%i,%i", pos.x, pos.y ,pos.z)
+	fs = fs:gsub("current_name", nodemeta)
+	fs = fs:gsub("context", nodemeta)
+
+	-- Change all ${} to their corresponding metadata values
+	local s, e
+	repeat
+		s, e = fs:find('%${.*}')
+		if s and e then
+			fs = fs:sub(1, s-1)..
+				minetest.formspec_escape(meta:get_string(fs:sub(s+2,e-1)))..
+				fs:sub(e+1)
+		end
+	until s == nil
+
+	-- Find node on_receive_fields
+	local ndef = minetest.registered_nodes[minetest.get_node(pos).name]
+
+	local context = get_context(playername)
+	context.node_pos = pos
+
+	if ndef and ndef.on_receive_fields then
+		context.on_receive_fields = ndef.on_receive_fields
+	end
+
+	-- Show formspec
+	minetest.show_formspec(playername, modname..':context_formspec', fs)
+end
+
+minetest.register_on_player_receive_fields(function(player, formname, fields)
+	if formname == modname..':context_formspec' then
+		local context = get_context(player)
+		if context == nil then
+			return
+		end
+
+		if context.on_receive_fields then
+			context.on_receive_fields(context.node_pos, '', fields, player)
+		end
+	end
+end)
+
+-- Specific functions
+
+local function get_nb_of_pages(context)
+	return math.ceil((#context.items + 1) / (style.lines * style.cols))
+end
+
+local function item_list_prepare(item_list)
+	local list = {}
+	for _, name in ipairs(item_list) do
+		local ndef = minetest.registered_items[name]
+		if ndef then
+			list[#list+1] = ndef
+		end
+	end
+	table.sort(list, function(a,b) return a.description < b.description end)
+	return list
+end
+
+local function show_item_list_formspec(player)
+	local context = get_context(player)
+	if context == nil then
+		return
+	end
+
+	local fs = 'size['..(style.colsize * style.cols)..','
+		..(style.linesize * style.lines + 1.3)..']'
+		..default.gui_bg..default.gui_bg_img..default.gui_slots
+		..'label[1,0;'..context.title..']'
+		..'button_exit['..(style.colsize * style.cols - 2)..','
+		..(style.linesize * style.lines + 0.7)..';2,1;close;Close]'
+
+	if context.page > 1 then
+	 	fs = fs..'button['..(style.colsize * style.cols - 2)..',-0.2;1,1;prev;<]'
```

Comment:
.git/rebase-apply/patch:386: space before tab in indent.

It's in between the two tabulators

Reply:
Removed